html/captive-portal/lib/captiveportal/PacketFence/Controller/Oauth2.pm

```diff
@@ -176,41 +193,50 @@ sub oauth2Result : Path : Args(1) {
     } elsif (lc($provider) eq 'windowslive') {
         $type = pf::Authentication::Source::WindowsLiveSource->meta->get_attribute(
             'type')->default;
+    } elsif (lc($provider) eq 'twitter') {
+        $type = pf::Authentication::Source::TwitterSource->meta->get_attribute(
+            'type')->default;
     }
     
     my $source = $profile->getSourceByType($type);
     if ($source) { 
-        # request a JSON response
-        my $h = HTTP::Headers->new( 'x-li-format' => 'json' );
-        $response = $token->get($source->{'protected_resource_url'}, $h ); 
-        if ($response->is_success) {
-            if ($provider eq 'linkedin'){
-                # response is sent as "email@example.com" with quotes
-                $pid = $response->content() ;
-                # remove the quotes
-                $pid =~ s/"//g;
-            }
-            else{
-                # Grab JSON content
-                my $json      = new JSON;
-                my $json_text = $json->decode($response->content());
-                if ($provider eq 'google' || $provider eq 'facebook') {
-                    $pid = $json_text->{email};
-                } elsif ($provider eq 'windowslive'){
-                    $pid = $json_text->{emails}->{account};
-                } elsif ($provider eq 'github'){
-                    # github doesn't provide email by default
-                    $pid = $json_text->{login}.'@github';
+        # in twitter, the username comes with the access token through our homemade lib
+        if($provider eq "twitter"){
```

Comment:
Put a space before and after a condtion and use single quotes for constant strings.